### DISCUSSION TITLE
Commands and filters auto-completion

### DISCUSSION THREAD

**Discussion by GeoffreyGx** [ppy/osu#30837] [Ideas]
It would be nice for new players and even long time players wanting to quickly find a command or a filter to have auto-completion for commands in chat and filters in the search bar.  I guess it would work a bit like minecraft with a popup above or below the textbox with a list of commands depending on what you've already typed, a description of the commands, and what arguments the command takes.

Here's a little concept I made on figma : 
![image](https://github.com/user-attachments/assets/949ca506-1f07-4fd8-be2d-8b0ddcf625d2)

**Comment by zium1337** [comment]
Also auto-suggest player names in MP lobbys would be nice
